tests/functional/timings.test.js

```diff
@@ -561,6 +561,80 @@ function runClsTests(loader) {
       t.end()
     }
   })
+
+  testDriver.test('cls only accumulates biggest session (short CLS session followed by long)', supportedCls, function (t, browser, router) {
+    const rumPromise = router.expectRum()
+    const loadPromise = browser
+      .safeGet(router.assetURL('cls-multiple-small-then-big.html', { loader: loader }))
+      .waitForConditionInBrowser('window.contentAdded === true', 10000)
+      .eval('window.allCls')
+
+    Promise.all([rumPromise, loadPromise])
+      .then(() => {
+        let timingsPromise = router.expectTimings()
+        let domPromise = browser.get(router.assetURL('/'))
+        return Promise.all([timingsPromise, domPromise, loadPromise])
+      })
+      .then(([timingsResult, domResult, loadResult]) => {
+        const {body, query} = timingsResult
+        const timings = querypack.decode(body && body.length ? body : query.e)
+        const timing = timings.find(t => t.name === 'lcp')
+        const cls = timing.attributes.find(a => a.key === 'cls')
+        console.log('CLS COULD BE ANY OF:', loadResult)
+        console.log('CLS SHOULD BE:', Math.max(...loadResult))
+        console.log('CLS ACTUALLY IS:', cls.value)
+        t.ok(cls.value >= 0, 'cls is a non-negative value')
+        t.ok(cls.value === Math.max(...loadResult), 'CLS is set to the largest CLS session')
+        t.equal(cls.type, 'doubleAttribute', 'cls is doubleAttribute')
+
+        t.end()
+      })
+      .catch(fail)
+
+    function fail (e) {
+      t.error(e)
+      t.end()
+    }
+  })
+
+  testDriver.test('cls only accumulates biggest session (long CLS session followed by short)', supportedCls, function (t, browser, router) {
+    const rumPromise = router.expectRum()
+    const loadPromise = browser
+      .safeGet(router.assetURL('cls-multiple-big-then-small.html', { loader: loader }))
+      .waitForConditionInBrowser('window.contentAdded === true', 10000)
+      .eval('window.allCls')
+
+    Promise.all([rumPromise, loadPromise])
+      .then(() => {
+        let timingsPromise = router.expectTimings()
+        let domPromise = browser.get(router.assetURL('/'))
+        return Promise.all([timingsPromise, domPromise, loadPromise])
+      })
+      .then(([timingsResult, domResult, loadResult]) => {
+        const {body, query} = timingsResult
+        const timings = querypack.decode(body && body.length ? body : query.e)
+        const load = timings.find(t => t.name === 'load')
```

Comment:
I am wondering how this is passing since the HTML page is waiting for the window load event to add shifting content (which would not show up on the load timing event). Ultimately, I think it would make sense to inspect the windowUnload timing event.

Reply:
`loadPromise` is doubly async here, which is why it is in the `Promise.all()` for a second time on `611`, and that is what is returning the `loadResult` array of CLS timings.  working with `wd` is certainly a little cumbersome to retrieve some of these basic properties, but this was just one way I found to pass that data along before the `browser.get('/')` was executed.  Feel free to change it for easier understanding!

Reply:
also this `lcpCls` test will need to be changed to match the conclusion we came to in our conversation, as it will no longer be the `largest` cls, you may have to find a different way to ensure it is the correct value.